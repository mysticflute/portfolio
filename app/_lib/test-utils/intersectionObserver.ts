import { jest } from '@jest/globals';

/**
 * Returns a mock implementation of window.IntersectionObserver, with the given
 * options and functions mocked with `jest.fn`.
 */
export function getMockIntersectionObserver() {
  return jest.fn((options: IntersectionObserverInit = {}) => {
    const mockObserver: IntersectionObserver = {
      root: options.root || null,
      rootMargin: options.rootMargin || '',
      thresholds: Array.isArray(options.threshold)
        ? options.threshold
        : [options.threshold ?? 0],
      observe: jest.fn(() => {}),
      unobserve: jest.fn(() => {}),
      disconnect: jest.fn(() => {}),
      takeRecords: jest.fn<() => IntersectionObserverEntry[]>(),
    };
    return mockObserver;
  });
}
