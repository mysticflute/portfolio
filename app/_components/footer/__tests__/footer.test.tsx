import {
  jest,
  beforeEach,
  afterEach,
  describe,
  expect,
  it,
} from '@jest/globals';
import { render, screen } from '@testing-library/react';
import { getMockIntersectionObserver } from '@/lib/test-utils/intersectionObserver';
import Footer from '../footer';

describe('footer', () => {
  let mockIntersectionObserver: jest.Mock<() => IntersectionObserver>;

  beforeEach(() => {
    mockIntersectionObserver = getMockIntersectionObserver();
    global.IntersectionObserver = mockIntersectionObserver;
  });

  afterEach(() => {
    mockIntersectionObserver.mockReset();
  });

  it('renders a copyright for the current year', () => {
    render(<Footer />);

    const currentYear = new Date().getFullYear();
    expect(
      screen.getByText(new RegExp(`copyright Â© ${currentYear}`, 'i')),
    ).toBeInTheDocument();
  });

  it('renders the expected set of links', () => {
    render(<Footer />);

    const allLinks = screen
      .getAllByRole<HTMLLinkElement>('link')
      .map(l => l.href);

    expect(allLinks).toMatchSnapshot();
  });
});
