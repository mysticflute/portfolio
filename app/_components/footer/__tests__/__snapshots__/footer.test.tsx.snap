// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders the expected set of links 1`] = `
[
  "http://localhost/",
  "https://twitter.com/mysticflute",
  "https://www.instagram.com/mysticflute/",
  "https://www.youtube.com/@nathandavidmcwilliams",
  "https://soundcloud.com/nathan_david_mcwilliams",
  "https://linktr.ee/nathanmcwilliams",
  "http://localhost/",
  "http://localhost/about",
  "http://localhost/#portfolio",
  "http://localhost/contact",
  "https://soundcloud.com/nathan_david_mcwilliams",
  "https://www.youtube.com/@nathandavidmcwilliams",
  "https://mysticflute.itch.io/",
  "https://twitter.com/mysticflute",
  "https://www.instagram.com/mysticflute/",
  "https://www.threads.net/@mysticflute",
  "https://bsky.app/profile/mysticflute.bsky.social",
  "https://linktr.ee/nathanmcwilliams",
  "mailto:spam@ndm.studio",
]
`;
